File: src/main/java/com/yuyan/imemodule/utils/expression/tokenizer/Token.java
package com.yuyan.imemodule.utils.expression.tokenizer;

/**
 * Abstract class for tokens used by exp4j to tokenize expressions
 */
public abstract class Token {
    public static final short TOKEN_NUMBER = 1;
    public static final short TOKEN_OPERATOR = 2;
    public static final short TOKEN_FUNCTION = 3;
    public static final short TOKEN_PARENTHESES_OPEN = 4;
    public static final short TOKEN_PARENTHESES_CLOSE = 5;
    public static final short TOKEN_VARIABLE = 6;
    public static final short TOKEN_SEPARATOR = 7;

    private final int type;

    Token(int type) {
        this.type = type;
    }

    public int getType() {
        return type;
    }

}

File: src/main/java/com/yuyan/imemodule/libs/expression/tokenizer/Token.kt
package com.yuyan.imemodule.libs.expression.tokenizer

/**
 * Abstract class for tokens used by exp4j to tokenize expressions
 */
abstract class Token internal constructor(@JvmField val type: Short) {
    companion object {
        const val TOKEN_NUMBER: Short = 1
        const val TOKEN_OPERATOR: Short = 2
        const val TOKEN_FUNCTION: Short = 3
        const val TOKEN_PARENTHESES_OPEN: Short = 4
        const val TOKEN_PARENTHESES_CLOSE: Short = 5
        const val TOKEN_VARIABLE: Short = 6
        const val TOKEN_SEPARATOR: Short = 7
    }
}
